design/src/rfcs/rfc0004_retry_behavior.md

```diff
@@ -0,0 +1,147 @@
+- By calling the `aws_config::ConfigLoader::retry_config(..)` method
+- By calling the retry_config method on a service-specific config
+
+The above list is in order of increasing precedence e.g. setting retry attempts with the `retry_config` method will override a value set by `AWS_MAX_ATTEMPTS`.
+
+## Setting an environment variable
+
+Here's an example app that logs your AWS user's identity
+
+```rust
+use aws_sdk_sts as sts;
+
+#[tokio::main]
+async fn main() -> Result<(), sts::Error> {
+    let config = aws_config::load_from_env().await;
+
+    let sts = sts::Client::new(&config);
+    let resp = sts.get_caller_identity().send().await?;
+    println!("your user id: {}", resp.user_id.unwrap_or_default());
+    Ok(())
+}
+```
+
+Then, in your terminal:
+
+```sh
+# Set the env var before running the example program
+export AWS_MAX_ATTEMPTS=5
+# Run the example program
+cargo run
+```
+
+## Calling a method on an AWS shared config
+
+Here's an example app that creates a shared config with custom retry behavior and then logs your AWS user's identity
+
+```rust
+use aws_sdk_sts as sts;
+use aws_types::config::Config;
+use aws_types::retry_config::RetryConfig;
+
+async fn create_config() -> aws_types::config::Config {
+    let retry_config = RetryConfig::builder().max_attempts(5).build();
+    let config = aws_config::from_env()
+        .retry_config(retry_config)
+        .load().await;
+
+    config
+}
+
+#[tokio::main]
+async fn main() -> Result<(), sts::Error> {
+    let config = create_config().await;
+
+    let sts = sts::Client::new(&config);
+    let resp = sts.get_caller_identity().send().await?;
+    println!("your user id: {}", resp.user_id.unwrap_or_default());
+    Ok(())
+}
+```
+
+## Calling a method on service-specific config
+
+Here's an example app that creates a service-specific config with custom retry behavior and then logs your AWS user's identity
+
+```rust
+use aws_sdk_sts as sts;
+use aws_types::config::Config;
+use aws_types::retry_config::RetryConfig;
+
+#[tokio::main]
+async fn main() -> Result<(), sts::Error> {
+    let config = aws_config::load_from_env().await;
+    let retry_config = RetryConfig::builder().max_attempts(5).build();
+    let sts_config = sts::config::Config::from(&config).retry_config(retry_config).build();
+
+    let sts = sts::Client::new(&sts_config);
+    let resp = sts.get_caller_identity().send().await?;
+    println!("your user id: {}", resp.user_id.unwrap_or_default());
+    Ok(())
+}
+```
+
+## Disabling retries
+
+Here's an example app that creates a service-specific config with custom retry behavior disabling retries and then logs your AWS user's identity
+
+```rust
+use aws_sdk_sts as sts;
+use aws_types::config::Config;
+use aws_types::retry_config::RetryConfig;
+
+#[tokio::main]
+async fn main() -> Result<(), sts::Error> {
+    let config = aws_config::load_from_env().await;
+    let retry_config = RetryConfig::builder().max_attempts(0).build();
```

Comment:
Since this is called `max_attempts`, it seems like there should be a lower bound of 1, whereas if it was called `max_retries`, then a value of 0 makes sense to disable retry. Is this a bound that should be validated?